File: leetcode/max_profit_II.cpp
using namespace std;
#include <iostream>
#include <vector>


class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profitFromPriceGain = 0;
        for (int i = 0; i < prices.size()-1; i++) {
            if (prices[i] < prices[i+1]) {
                profitFromPriceGain += prices[i+1] - prices[i];
            }
        }
        return profitFromPriceGain;
    }
};

int main() {
    Solution s;
    vector<int> prices = {7, 1, 5, 3, 6, 4};
    cout << s.maxProfit(prices) << endl;
}

File: leetcode/max_profit_II.py
from typing import List 


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit_from_price_gain = 0
        for idx in range(len(prices) - 1):
            if prices[idx] < prices[idx + 1]:
                profit_from_price_gain += (prices[idx + 1] - prices[idx])
        return profit_from_price_gain
    
if __name__ == "__main__":
    s = Solution()  
    # prices = [7, 1, 5, 3, 6, 4]
    # prices = [1,2,3,4,5]
    prices = [7,6,4,3,1]
    print(s.maxProfit(prices))

